literalai/api/__init__.py

```diff
@@ -2593,16 +2655,34 @@ async def get_prompt(
         name: Optional[str] = None,
         version: Optional[int] = None,
     ) -> Prompt:
-        sync_api = LiteralAPI(self.api_key, self.url)
-        if id:
-            return await self.gql_helper(*get_prompt_helper(sync_api, id=id))
-        elif name:
-            return await self.gql_helper(
-                *get_prompt_helper(sync_api, name=name, version=version)
-            )
-        else:
+        if not (id or name):
             raise ValueError("Either the `id` or the `name` must be provided.")
 
+        sync_api = LiteralAPI(self.api_key, self.url)
+        cached_prompt = self.prompt_cache.get(id, name, version)
+        timeout = 1 if cached_prompt else None
```

Comment:
you could move the cache logic in the get_prompt_helper to avoid duplicating it for the sync/async versions.